src/main/java/eu/yals/ui/HomeView.java

```diff
@@ -0,0 +1,442 @@
+package eu.yals.ui;
+
+import com.github.appreciated.app.layout.annotations.Caption;
+import com.github.appreciated.app.layout.annotations.Icon;
+import com.vaadin.flow.component.ClickEvent;
+import com.vaadin.flow.component.board.Board;
+import com.vaadin.flow.component.board.Row;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.button.ButtonVariant;
+import com.vaadin.flow.component.dependency.StyleSheet;
+import com.vaadin.flow.component.html.*;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.component.notification.Notification;
+import com.vaadin.flow.component.notification.NotificationVariant;
+import com.vaadin.flow.component.orderedlayout.FlexComponent;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.data.value.ValueChangeMode;
+import com.vaadin.flow.router.PageTitle;
+import com.vaadin.flow.router.Route;
+import com.vaadin.flow.spring.annotation.SpringComponent;
+import com.vaadin.flow.spring.annotation.UIScope;
+import eu.yals.Endpoint;
+import eu.yals.json.StoreRequestJson;
+import eu.yals.services.overall.OverallService;
+import eu.yals.utils.AppUtils;
+import kong.unirest.HttpResponse;
+import kong.unirest.JsonNode;
+import kong.unirest.Unirest;
+import kong.unirest.json.JSONException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.vaadin.olli.ClipboardHelper;
+
+@Slf4j
+@SpringComponent
+@UIScope
+@StyleSheet("home_view.css")
+@Route(value = Endpoint.UI.HOME_PAGE, layout = AppView.class)
+@Caption("Home")
+@Icon(VaadinIcon.HOME)
+@PageTitle("Link shortener for friends")
+public class HomeView extends VerticalLayout {
+  private static final String TAG = "[Front Page]";
+
+  private final Board board = new Board();
+  private final Row firstRow = new Row();
+  private final Row mainRow = new Row();
+  private final Row overallRow = new Row();
+  private final Row resultRow = new Row();
+  private final Row qrCodeRow = new Row();
+
+  private final OverallService overallService;
+  private final AppUtils appUtils;
+
+  private TextField input;
+  private Button submitButton;
+  Anchor shortLink;
+  ClipboardHelper clipboardHelper;
+  Image qrCode;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=5392475)